### PR TITLE
Remove deprecated fields

### PR DESCRIPTION
**Pull request by mziolekda**

### CODE REVIEW

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto [hunk 1]
```diff
@@ -219,23 +219,11 @@ message DisclosedContract {
...
-    // of a ``com.daml.ledger.api.v1.CreatedEvent``.
-    // Deprecated in favor of `created_event_blob`
-    google.protobuf.Any create_arguments_blob = 5 [deprecated = true];
-  }
+  reserved 4; // removed metadata
 
-  // The contract metadata from the create event.
-  // Deprecated in favor of `created_event_blob`
-  // Optional
-  ContractMetadata metadata = 4 [deprecated = true];
```
Comment (cocreature) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:238]: You can also kill the masseg itself no?

Comment (meiersi-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:238]: We'd be loosing extensibility wouldn't we?

Comment (cocreature) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:238]: by killing an unused `ContractMetadata` field? I don't see how that loses extensibilityt.

Comment (meiersi-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:238]: Ah sorry, I've thought you want to remove the `DisclosedContract` message, as it now only contains a single field.

Comment (mziolekda) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:238]: Yes, yes. Canton repo compilation told me about this too.

Comment (mziolekda) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:238]: Actually, I have learned that the metadata is used in canton repair tooling and is needed for the old exports

File: ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_15/ExplicitDisclosureIT.scala [hunk 2]
```diff
@@ -447,45 +445,6 @@ final class ExplicitDisclosureIT extends LedgerTestSuite {
...
-            "The submitted command has invalid arguments: DisclosedContract.arguments or DisclosedContract.metadata cannot be set together with DisclosedContract.create_event_payload"
-          ),
-          checkDefiniteAnswerMetadata = true,
-        )
-      }
-  })
-
   test(
     "EDRejectOnPayloadNotSet",
     "Submission is rejected when the disclosed contract payload is not set",
```
Comment (tudor-da) [inline ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_15/ExplicitDisclosureIT.scala:450]:
```suggestion
    "Submission is rejected when the disclosed contract event blob is not set",
```

File: ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_15/ExplicitDisclosureIT.scala [hunk 3]
```diff
@@ -523,7 +480,7 @@ final class ExplicitDisclosureIT extends LedgerTestSuite {
           LedgerApiErrors.RequestValidation.MissingField,
           Some(
             // TODO ED: Change assertion to missing create_event_payload once the deprecated fields are removed
```
Comment (tudor-da) [inline ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_15/ExplicitDisclosureIT.scala:482]:
```suggestion
```

**Review by tudor-da — APPROVED** [review]
lgtm! Left two small comments, but not blocking for this PR

**Comment by mziolekda** [comment]
Cool, I will merge, and correct in a follow-up because CI on this PR is awfully expensive. Due to it touching the grpc definitions, it rebuilds todo el mundo

**Pull request merged**